core/src/main/java/hudson/Functions.java

```diff
@@ -1571,7 +1572,16 @@ public static String toCCStatus(Item i) {
      * Checks if the current user is anonymous.
      */
     public static boolean isAnonymous() {
-        return Jenkins.getAuthentication() instanceof AnonymousAuthenticationToken;
+        return isAnonymous(Jenkins.getAuthentication());
+    }
+
+    /**
+     * Checks if the given authentication is anonymous.
+     * @since TODO
```

Comment:
Probably belongs elsewhere, such as in `ACL` perhaps.

`Functions` is _not_ a general-purpose utility class. It is bound to `h` in Jelly and I think also Groovy views, and is specifically intended as a holder for bits of Java logic that Stapler views would frequently need to invoke (and that they should not inline).

Reply:
Comment still stands. Minimally make this method restricted.

Reply:
Exactly what Oleg adviced me to do for the moment. We did not find a better place at the moment. The `ACL` class just had a notion of anonymous authentication like `Jenkins` but no method associated.